diff --git a/src/kitemviews/kfileitemlistwidget.cpp b/src/kitemviews/kfileitemlistwidget.cpp
index 37b115cc3..a7bfff75a 100644
--- a/src/kitemviews/kfileitemlistwidget.cpp
+++ b/src/kitemviews/kfileitemlistwidget.cpp
@@ -56,6 +56,9 @@ QString KFileItemListWidgetInformant::roleText(const QByteArray &role, const QHa
     // use a hash + switch for a linear runtime.
 
     auto formatDate = [formatter, local, forUsageAs](const QDateTime &time) {
+        
+        return time.toString("yyyy-MM-dd HH:mm:ss t");
+        
         if (ContentDisplaySettings::useShortRelativeDates()) {
             return formatter.formatRelativeDateTime(time,
                                                     forUsageAs == KStandardItemListWidgetInformant::ForUsageAs::DisplayedText ? QLocale::ShortFormat
diff --git a/src/views/tooltips/dolphinfilemetadatawidget.cpp b/src/views/tooltips/dolphinfilemetadatawidget.cpp
index ba068d3f4..7c0f2fec0 100644
--- a/src/views/tooltips/dolphinfilemetadatawidget.cpp
+++ b/src/views/tooltips/dolphinfilemetadatawidget.cpp
@@ -56,6 +56,7 @@ DolphinFileMetaDataWidget::DolphinFileMetaDataWidget(QWidget *parent)
     textLayout->addWidget(m_fileMetaDataWidget);
     textLayout->setAlignment(m_name, Qt::AlignCenter);
     textLayout->setAlignment(m_fileMetaDataWidget, Qt::AlignLeft);
+    m_fileMetaDataWidget->setMinimumWidth(fontMetrics.averageCharWidth() * 45);
     // Assure that the text-layout gets top-aligned by adding a stretch.
     // Don't use textLayout->setAlignment(Qt::AlignTop) instead, as this does
     // not work with the heightForWidth()-size-hint of m_fileMetaDataWidget
