--- kf6-kio-6.13.0.orig/src/widgets/kpropertiesdialogbuiltin_p.cpp
+++ kf6-kio-6.13.0/src/widgets/kpropertiesdialogbuiltin_p.cpp
@@ -476,21 +476,21 @@ KFilePropsPlugin::KFilePropsPlugin(KProp
     if (!d->bMultiple) {
         QLocale locale;
         if (const QDateTime dt = firstItem.time(KFileItem::CreationTime); !dt.isNull()) {
-            d->m_ui->createdTimeLabel->setText(locale.toString(dt, QLocale::LongFormat));
+            d->m_ui->createdTimeLabel->setText(dt.toString(QString::fromUtf8("yyyy-MM-dd HH:mm:ss t")));
         } else {
             d->m_ui->createdTimeLabel->hide();
             d->m_ui->createdTimeLabel_Left->hide();
         }
 
         if (const QDateTime dt = firstItem.time(KFileItem::ModificationTime); !dt.isNull()) {
-            d->m_ui->modifiedTimeLabel->setText(locale.toString(dt, QLocale::LongFormat));
+            d->m_ui->modifiedTimeLabel->setText(dt.toString(QString::fromUtf8("yyyy-MM-dd HH:mm:ss t")));
         } else {
             d->m_ui->modifiedTimeLabel->hide();
             d->m_ui->modifiedTimeLabel_Left->hide();
         }
 
         if (const QDateTime dt = firstItem.time(KFileItem::AccessTime); !dt.isNull() && hasAccessTime) {
-            d->m_ui->accessTimeLabel->setText(locale.toString(dt, QLocale::LongFormat));
+            d->m_ui->accessTimeLabel->setText(dt.toString(QString::fromUtf8("yyyy-MM-dd HH:mm:ss t")));
         } else {
             d->m_ui->accessTimeLabel->hide();
             d->m_ui->accessTimeLabel_Left->hide();
